csharp/GildedRoseTest.cs

```diff
@@ -7,12 +7,185 @@ namespace csharp
     public class GildedRoseTest
```

Comment:
It's useful to split each test into 
// given // when // then blocks, where given contains app init & adding items, when contains changing quality, then containing asserts

Also,